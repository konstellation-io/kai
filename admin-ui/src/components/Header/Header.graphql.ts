import gql from 'graphql-tag';
import { User } from '../../graphql/models';

export interface GetUserEmailResponse {
  me: User;
}

export const GET_USER_EMAIL = gql`
  query GetUserEmail {
    me {
      email
    }
  }
`;
