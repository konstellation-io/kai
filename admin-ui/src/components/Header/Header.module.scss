.container {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  flex-shrink: 0;
  padding: 0px 8px 0 13px;
  width: 100%;
  height: 40px;
  background-color: black;
  white-space: nowrap;
}

.konstellationsIcon {
  height: 18px;
  margin-right: 20px;
}

.customHeaderElements {
  height: 100%;
  display: flex;
  flex-direction: row-reverse;
  flex-grow: 1;
  align-items: center;

  flex-wrap: wrap;
  overflow: hidden;
}

.splash {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background-color: black;
  z-index: 100;
}
