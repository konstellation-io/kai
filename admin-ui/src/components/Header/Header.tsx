import { get } from 'lodash';

import React, { FunctionComponent, ReactElement } from 'react';

import Settings from '../../components/Settings/Settings';

import { loader } from 'graphql.macro';
import { useQuery } from '@apollo/react-hooks';

import { GetUserEmail } from '../../graphql/queries/types/GetUserEmail';

import styles from './Header.module.scss';

const GetUserEmailQuery = loader('../../graphql/queries/getUserEmail.graphql');

type Props = {
  children?: ReactElement | ReactElement[] | null;
};
const Header: FunctionComponent<Props> = ({ children }) => {
  const { data, loading } = useQuery<GetUserEmail>(GetUserEmailQuery);

  if (loading)
    return <div className={styles.splash} data-testid="splashscreen" />;

  const username: string = get(data, 'me.email');

  return (
    <header className={styles.container} data-testid="app-header">
      <img
        className={styles.konstellationsIcon}
        src={'/img/brand/konstellation.png'}
        alt="konstellation logo"
      />
      <div className={styles.customHeaderElements}>{children}</div>
      <Settings label={username} />
    </header>
  );
};

export default Header;
