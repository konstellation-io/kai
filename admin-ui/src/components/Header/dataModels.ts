import gql from 'graphql-tag';

export const GET_USERNAME = gql`
  query GetUserName {
    me {
      email
    }
  }
`;
