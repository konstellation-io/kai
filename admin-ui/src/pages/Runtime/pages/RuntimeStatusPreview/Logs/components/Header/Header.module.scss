@import '../../../../../../../styles/mixins';
@import '../../../../../../../styles/funcs';
@import '../../../../../../../styles/colors';

.container {
  display: flex;
  align-items: center;
}

.statusCircle {
  width: 8px;
  height: 8px;
  margin: 0 16px;
  border-radius: 50%;

  @each $status in 'PUBLISHED' {
    background-color: get-color-prop($status);
  }
}

.runtime {
  @include font-body;
  color: $font-color-white;
}

.version {
  @include font-body;
  color: $font-color-regular;

  margin-left: 16px;
}

.buttons {
  display: flex;
  flex-direction: row;
  opacity: 0;
  transition: opacity ease 0.4s;
  margin: 4px 8px 0 auto;

  &.opened {
    opacity: 1;
  }

  div {
    margin-left: 8px;
    cursor: pointer;

    &:hover {
      color: $font-color-white;
    }
  }
}

.stickBottom.active {
  color: $font-color-white;

  &:hover {
    color: $font-color-light;
  }
}
