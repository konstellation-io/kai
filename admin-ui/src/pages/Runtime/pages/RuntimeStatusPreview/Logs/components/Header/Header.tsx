import React from 'react';

import IconClose from '@material-ui/icons/Close';
import IconStickBottom from '@material-ui/icons/VerticalAlignBottom';

import cx from 'classnames';
import styles from './Header.module.scss';

type Props = {
  runtimeName: string;
  versionName: string;
  closeLogs: () => void;
  opened: boolean;
  stickToBottom: boolean;
  toggleStickToBottom: () => void;
};
function Header({
  runtimeName,
  versionName,
  closeLogs,
  opened,
  stickToBottom,
  toggleStickToBottom
}: Props) {
  function scrollToBottom() {
    const listContainer = document.getElementById('VersionLogsListContainer');
    if (listContainer) {
      listContainer.scrollTo({
        top: listContainer.scrollHeight,
        behavior: 'smooth'
      });
    }
  }

  return (
    <div className={styles.container}>
      <div className={styles.statusCircle} />
      <span className={styles.runtime}>{runtimeName}</span>
      <span className={styles.version}>{versionName}</span>
      <div
        className={cx(styles.buttons, {
          [styles.opened]: opened
        })}
      >
        <div
          className={cx(styles.stickBottom, { [styles.active]: stickToBottom })}
          onClick={toggleStickToBottom}
        >
          <IconStickBottom className="icon-regular" />
        </div>
        <div onClick={closeLogs}>
          <IconClose className="icon-regular" />
        </div>
      </div>
    </div>
  );
}

export default Header;
