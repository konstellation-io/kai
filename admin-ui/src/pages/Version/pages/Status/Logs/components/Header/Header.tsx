import React from 'react';

import IconOpen from '@material-ui/icons/ExpandLess';
import IconClose from '@material-ui/icons/ExpandMore';
import IconStickBottom from '@material-ui/icons/VerticalAlignBottom';
import IconClear from '@material-ui/icons/DeleteOutline';
import IconLogs from '@material-ui/icons/ListAlt';

import { useApolloClient } from '@apollo/react-hooks';

import cx from 'classnames';
import styles from './Header.module.scss';

type Props = {
  togglePanel: () => void;
  opened: boolean;
  stickToBottom: boolean;
  toggleStickToBottom: () => void;
};
function Header({
  togglePanel,
  opened,
  stickToBottom,
  toggleStickToBottom
}: Props) {
  const client = useApolloClient();

  function clearLogs(): void {
    client.writeData({ data: { logs: [] } });
  }

  const Icon = opened ? IconClose : IconOpen;

  return (
    <div
      className={cx(styles.container, {
        [styles.opened]: opened
      })}
    >
      <div className={styles.title}>
        <IconLogs className="icon-regular" />
        <span>Logs console</span>
      </div>
      <div className={styles.buttons}>
        <div onClick={clearLogs}>
          <IconClear className="icon-regular" />
        </div>
        <div
          className={cx(styles.stickBottom, { [styles.active]: stickToBottom })}
          onClick={toggleStickToBottom}
        >
          <IconStickBottom className="icon-regular" />
        </div>
        <div onClick={togglePanel}>
          <Icon className="icon-regular" />
        </div>
      </div>
    </div>
  );
}

export default Header;
