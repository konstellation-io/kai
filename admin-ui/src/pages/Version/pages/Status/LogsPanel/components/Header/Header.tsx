import React from 'react';

import IconOpen from '@material-ui/icons/ExpandLess';
import IconClose from '@material-ui/icons/ExpandMore';
import * as ICONS from '../../../../../../../constants/icons';

import cx from 'classnames';
import styles from './Header.module.scss';
import SvgIcon from '@material-ui/core/SvgIcon';

type Props = {
  togglePanel: () => void;
  opened: boolean;
};
function Header({ togglePanel, opened }: Props) {
  const Icon = opened ? IconClose : IconOpen;

  return (
    <div
      className={cx(styles.container, {
        [styles.opened]: opened
      })}
    >
      <div className={styles.title}>
        <SvgIcon className="icon-small">
          <path d={ICONS.TERMINAL} />
        </SvgIcon>
        <span>LOGS CONSOLE</span>
      </div>
      <div className={styles.buttons}>
        <div onClick={togglePanel}>
          <Icon className="icon-regular" />
        </div>
      </div>
    </div>
  );
}

export default Header;
