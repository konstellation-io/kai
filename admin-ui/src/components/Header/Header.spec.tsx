import React from 'react';
import { render } from '@testing-library/react';
import Header from './Header';

it('Render Header without crashing', () => {
  const { container } = render(<Header />);
  expect(container).toMatchSnapshot();
});
