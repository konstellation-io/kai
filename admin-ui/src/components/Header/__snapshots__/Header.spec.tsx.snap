// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header without crashing 1`] = `
<div>
  <header
    class="container"
  >
    <img
      alt="konstellation logo"
      class="konstellationsIcon"
      src="/img/brand/konstellation.svg"
    />
    <div
      class="customHeaderElements"
    />
    <div
      class="container"
      data-testid="settingsContainer"
    >
      <div
        class="label"
      >
        mariano@intelygenz.com
        <div
          class="arrow"
        />
      </div>
      <div
        class="options"
        data-testid="settingsContent"
        style="max-height: 0;"
      >
        <div
          class="btn grey left label"
          style="padding-left: 20%; height: 40px; line-height: 40px;"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            style="font-size: 1rem;"
            viewBox="0 0 24 24"
          >
            <path
              d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <span>
            LOGOUT
          </span>
        </div>
      </div>
    </div>
  </header>
</div>
`;
